activerecord/lib/active_record/associations/belongs_to_association.rb

```diff
@@ -72,23 +67,24 @@ def decrement_counters_before_last_save
           model_was = klass
         end
 
-        foreign_key_was = owner.attribute_before_last_save(reflection.foreign_key)
+        foreign_key_was = foreign_keys.map { |fk| owner.attribute_before_last_save(fk) }
 
-        if foreign_key_was && model_was < ActiveRecord::Base
+        if foreign_key_was && model_was && model_was < ActiveRecord::Base
```

Comment:
`foreign_key_was` is going to be always true here, no?

Reply:
```suggestion
        if foreign_key_was.any? && model_was && model_was < ActiveRecord::Base
```

True, I think because `attribute_before_last_save` can return the value or nil(?) then this should work.